### PR TITLE
Write base64 screenshot data with proper encoding.

### PR DESCRIPTION
**Pull request by negiDharmendra**

### CODE REVIEW

File: src/screenshot.js [hunk 1]
```diff
@@ -45,26 +45,25 @@ function getScreenshotFunc() {
...
+          writeScreenshotWithErrorHandling(res, screenshotFile, resolve);
         }
       });
     };
   }
   return defaultScreenshotWriter;
 }
 
-function errorHandling(action) {
+function writeScreenshotWithErrorHandling(data, screenshotFile, resolver) {
```
Comment (sriv) [inline src/screenshot.js]: `writeScreenshotWithErrorHandling` suggests that there is some error handling, but I don't see any error handling. can we rename this?

File: test/screenshot-test.js [hunk 2]
```diff
@@ -44,65 +44,105 @@ describe("screentshot.capture", function () {
...
+        });
       });
-    });
 
-    it("Should capture screentshot with sync function", function (done) {
-      sandbox.stub(process.hrtime, "bigint").returns(6767787989089);
-      const screenShotFile = "screenshot-6767787989089.png";
-      global.gauge = { screenshotFn: screenshotFunction };
-      const writeFileSyncStub = sandbox.stub(fs, "writeFileSync");
+      it("Should capture screentshot with sync function", function (done) {
```
Comment (sriv) [inline test/screenshot-test.js]: are the indents expected?

File: test/screenshot-test.js [hunk 3]
```diff
@@ -44,65 +44,105 @@ describe("screentshot.capture", function () {
...
-    it("Should capture screentshot with async function", function (done) {
-      const screenShotFile = "screenshot-file-1.png";
-      global.gauge = { customScreenshotWriter: function() {
-        return Promise.resolve(screenShotFile);
-      } };
+      describe("when data is in bytes", () => {
+        it("Should capture screentshot with async function", function (done) {
+          sandbox.stub(process.hrtime, "bigint").returns(6767787989089);
+          const screenShotFile = "screenshot-6767787989089.png";
+          global.gauge = { screenshotFn: () => Promise.resolve(Buffer.from("screentshot")) };
```
Comment (sriv) [inline test/screenshot-test.js]: should use `async` / `await` in the syntax or rename the spec to mention "function returning a promise"

File: test/screenshot-test.js (hunk 2 above)
Comment (sriv) [reply test/screenshot-test.js]: false alarm

**Comment by Debashis9012** [comment]
fixed #295

**Comment by Debashis9012** [comment]
This has been tested and working as excepted.

**Pull request merged**